### PR TITLE
PIMS-874 Cleaning up documentation and DB configuration process.

### PR DESCRIPTION
**Pull request by Fosol**
Corrected a number of issues with documentation related to setting up the database and using the `dotnet-ef` CLI.

### CODE REVIEW

File: backend/Data/PIMSContextFactory.cs [hunk 1]
```diff
@@ -11,16 +11,17 @@ public class PIMSContextFactory : IDesignTimeDbContextFactory<PIMSContext>
...
         {
+            DotNetEnv.Env.Load();
             string environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
 
             var config = new ConfigurationBuilder()
                 .SetBasePath(Directory.GetCurrentDirectory())
                 .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
                 .AddJsonFile($"appsettings.{environment}.json", optional: true, reloadOnChange: true)
+                .AddEnvironmentVariables()
                 .AddJsonFile("connectionstrings.json", optional: true, reloadOnChange: true)
```
Comment (Fosol) [inline backend/Data/PIMSContextFactory.cs]: You can now have configuration for the `dotnet-ef` CLI that doesn't interfere with your `.env` and docker containers.

File: backend/pims.api.csproj [hunk 2]
```diff
@@ -6,6 +6,7 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="7.0.0" />
+    <PackageReference Include="DotNetEnv" Version="1.4.0" />
```
Comment (Fosol) [inline backend/pims.api.csproj]: Allows the `dotnet-ef` CLI to use `.env` files for configuration if we want them to.

File: backend/README.md [hunk 3]
```diff
@@ -1,26 +1,80 @@
...
+The `Keycloak__Secret` should be the value provided by KeyCloak. Currently this value can remain blank.
+
 ## Secret Management
 
-To keep private keys and user secrets out of source code use the **user-secrets** management tool.
+If you want to keep private keys and user secrets out of source code use the **user-secrets** management tool.
+
+> `dotnet user-secrets init`
+
+> `dotnet user-secrets set "ConnectionStrings:GeoSpatial" "Host=host.docker.internal;Port=5433;Username=sa;Password=[password];Database=geospatial;"`
```
Comment (asanchezr) [inline backend/README.md]: Could you please change the above connection string to match SQL Server format? I believe MSSQL doesn't support `Host` like postgres

File: backend/README.md [hunk 4]
```diff
@@ -57,3 +103,73 @@ Generate the SQL for the migration.
...
+
+To add a new database code migration do the following;
+
+Go to the `/backend` folder. Enter the name of the migration you want to create `[name]`.
+
+> `dotnet ef migrations add [name]`
+
+You should then edit the migration you created `[20200204191656_name].cs` to inherit from our `SeedMigration` class. This is to enable running SQL scripts during migration, for either complex db changes, or seed data.
+
+```
```
Comment (asanchezr) [inline backend/README.md]: /nit Please add [```csharp] for proper syntax highlighting. Thanks!

File: backend/README.md [hunk 5]
```diff
@@ -57,3 +103,73 @@ Generate the SQL for the migration.
...
+        }
+    }
+}
+```
+
+Any script that you want to run must be additionally included as build content in the project file.
+
+Edit the `Pims.Api.csproj` file and add the `<Content>` with a `<CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>` (see below example).
+
+```
```
Comment (asanchezr) [inline backend/README.md]: please add [```xml]

**Pull request merged**